scenes/test_string.txt

```diff
@@ -144,28 +144,28 @@
 *gosub test_start "EXTRACT_TOKENS #1"
 *set str_buffer "VAL=TRUE|VAL=FALSE|VAL=TRUE"
 *gosub_scene cslib_string extract_tokens str_buffer "|" "tokens"
-*gosub test_finish 3 cslib_ret
-*gosub test_finish "VAL=TRUE" tokens_1
-*gosub test_finish "VAL=FALSE" tokens_2
-*gosub test_finish "VAL=TRUE" tokens_3
+*gosub test_assert_equal 3 cslib_ret
+*gosub test_assert_equal "VAL=TRUE" tokens_1
+*gosub test_assert_equal "VAL=FALSE" tokens_2
+*gosub test_assert_equal "VAL=TRUE" tokens_3
 *comment The rest of the array is still empty.
-*gosub test_finish "" tokens_4
-*gosub test_finish "" tokens_5
-*gosub test_finish 5 tokens_max
+*gosub test_assert_equal "" tokens_4
+*gosub test_assert_equal "" tokens_5
+*gosub test_assert_equal 5 tokens_max
 *gosub test_finish 3 tokens_count
 
 *comment Extract 2 tokens into an array of 5. Remaining entries are reset.
 *gosub test_start "EXTRACT_TOKENS #2"
 *set str_buffer "VAL=HERE|VAL=THERE"
 *gosub_scene cslib_string extract_tokens str_buffer "|" "tokens"
-*gosub test_finish 2 cslib_ret
-*gosub test_finish "VAL=HERE" tokens_1
-*gosub test_finish "VAL=THERE" tokens_2
+*gosub test_assert_equal 2 cslib_ret
+*gosub test_assert_equal "VAL=HERE" tokens_1
+*gosub test_assert_equal "VAL=THERE" tokens_2
 *comment The rest of the array is wiped clean.
-*gosub test_finish "" tokens_3
-*gosub test_finish "" tokens_4
-*gosub test_finish "" tokens_5
-*gosub test_finish 5 tokens_max
+*gosub test_assert_equal "" tokens_3
+*gosub test_assert_equal "" tokens_4
+*gosub test_assert_equal "" tokens_5
+*gosub test_assert_equal 5 tokens_max
```

Comment:
I assume you meant to check _count here? Checking _max seems redundant (it should never change).

Reply:
I'm checking both, count is the next line. Better safe than sorry.